modules/library/data/src/main/java/org/geotools/data/store/ContentFeatureSource.java

```diff
@@ -425,6 +427,35 @@ public final ReferencedEnvelope getBounds(Query query) throws IOException {
         } else {
             bounds = getBoundsInternal(query);
         }
+        // reprojection
+        if ( !canReproject() ) {
+            CoordinateReferenceSystem sourceCRS = query.getCoordinateSystem();
+            CoordinateReferenceSystem targetCRS = query.getCoordinateSystemReproject();
+            CoordinateReferenceSystem nativeCRS = getSchema().getCoordinateReferenceSystem();
```

Comment:
Nice clear code @tbarsballe, I am impressed you even updated the date on these files.